usaspending_api/search/v2/views/spending_by_award.py

```diff
@@ -0,0 +1,259 @@
+            sort_filters = [award_contracts_mapping[self.sort]]
+        elif set(filters["award_type_codes"]) <= set(loan_type_mapping):  # loans
+            sort_filters = [loan_award_mapping[self.sort]]
+        elif set(filters["award_type_codes"]) <= set(idv_type_mapping):  # idvs
+            sort_filters = [award_idv_mapping[self.sort]]
+        else:  # assistance data
+            sort_filters = [non_loan_assistance_award_mapping[self.sort]]
+
+        # Explictly set NULLS LAST in the ordering to encourage the usage of the indexes
+        if self.sort == "Award ID":
+            if self.order == "desc":
+                queryset = queryset.order_by(
+                    F("piid").desc(nulls_last=True), F("fain").desc(nulls_last=True), F("uri").desc(nulls_last=True)
+                ).values(*list(values))
+            else:
+                queryset = queryset.order_by(
+                    F("piid").asc(nulls_last=True), F("fain").asc(nulls_last=True), F("uri").asc(nulls_last=True)
+                ).values(*list(values))
+        elif self.order == "desc":
+            queryset = queryset.order_by(F(sort_filters[0]).desc(nulls_last=True)).values(*list(values))
+        else:
+            queryset = queryset.order_by(F(sort_filters[0]).asc(nulls_last=True)).values(*list(values))
+
+        limited_queryset = queryset[self.lower_bound : self.upper_bound]
+        has_next = len(limited_queryset) > self.limit
+
+        results = []
+        for award in limited_queryset[: self.limit]:
+            row = {"internal_id": award["award_id"]}
+
+            if award["type"] in loan_type_mapping:
+                for field in self.fields:
+                    row[field] = award.get(loan_award_mapping.get(field))
+            elif award["type"] in non_loan_assistance_type_mapping:
+                for field in self.fields:
+                    row[field] = award.get(non_loan_assistance_award_mapping.get(field))
+            elif award["type"] in idv_type_mapping:
+                for field in self.fields:
+                    row[field] = award.get(award_idv_mapping.get(field))
+            elif award["type"] in contract_type_mapping:
+                for field in self.fields:
+                    row[field] = award.get(award_contracts_mapping.get(field))
+
+            if "Award ID" in self.fields:
+                for id_type in ["piid", "fain", "uri"]:
+                    if award[id_type]:
+                        row["Award ID"] = award[id_type]
+                        break
+            results.append(row)
+
+        return {"limit": self.limit, "results": results, "page_metadata": {"page": self.page, "hasNext": has_next}}
+
+    def _handle_subawards(self, filters):
+        """ sub-award logic for SpendingByAward"""
+        # Test to ensure the award types are a proper subset of at least one group (contracts or grants)
+        if set(filters["award_type_codes"]) - set(procurement_type_mapping.keys()) and set(
```

Comment:
Personally a fan of using `set_a.difference(set_b)` instead of `set_a - set_b` because it reads more clear.